app/src/neptune_eye/object_detection/yolo_object_detection.py

```
@@ -0,0 +1,174 @@
+    def _get_nvidia_gpu_model_path(self) -> str:
+        """Get model path for NVIDIA GPU inference.
+        """
+        precision_suffix = "16fp" if self.half_precision else "32fp"
+        
+        model_paths = {
+            YoloModelSize.YOLO11N: f"engine/yolo11n_{precision_suffix}.engine",
+            YoloModelSize.YOLO11S: f"engine/yolo11s_{precision_suffix}.engine",
+        }
+
+        if self.model_size not in model_paths:
+            raise RuntimeError(f"Unsupported model size for NVIDIA GPU: {self.model_size}")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)